geo/Geoserver.py

```diff
@@ -393,6 +393,121 @@ def create_workspace(self, workspace: str):
         except Exception as e:
             raise e
 
+    def create_featured_layers_rule(self, workspacePattern: str = '*', permission: str = 'r', role: str = None):
+        """
+        Create a new security rule for either all workspaces or a subset of them based on the provided pattern.
+
+        Parameters
+        ----------
+        workspacePattern : str
+            workspace pattern. Apply the security rule to either all workspaces (using *) or 
+            a subset of them where pattern matches (contains) the workspace name.
+        permission : str
+            [r|w|a] is a placeholder for the permission type. 
+            r read-only access, w write access and a indicates full (read and write) access.
+        role : str
+            rol name
+        """
+
+        if role is None:
+            raise Exception("You must provide a role name")
+
+        workspacesRawData = self.get_workspaces()
+
+        if workspacesRawData == None:
+            raise Exception("There is not workpaces available")
+
+        workspaceList = workspacesRawData['workspaces']['workspace']
+
+        if workspacePattern != '*':
+            workspaceList = [
+                workspace for workspace in workspaceList if workspacePattern in workspace['name']]
+
+        if len(workspaceList) == 0:
+            raise Exception("The pattern does not match any workspace")
+
+        for workspace in workspaceList:
+            res = self.create_update_layer_rule(
+                workspace['name'], '*', permission, role)
+            print(res)
```

Comment:
This can end up printing too many messages, but you can leave it. The thing is `create_featured_layers_rule` is still returning nothing

Reply:
Now returns the number of created and updated rules